pkg/provider/provider.go

```diff
@@ -538,18 +538,35 @@ func (k *kubeProvider) Diff(
 		oldInputs.SetNamespace("")
 		newInputs.SetNamespace("")
 	}
+	if oldInputs.GroupVersionKind().Empty() {
```

Comment:
Why is this necessary?

Reply:
If we have an empty old input set, the old input set will not have a GVK. `openapi.PatchForResourceUpdate` looks at the old input set to determine the GVK it will use to lookup patch meta. This ensures that the old inputs have a GVK that can be used by `openapi.PatchForResourceUpdate`.